core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/ApplyTableScanRedirection.java

```diff
@@ -87,6 +88,14 @@ public Result apply(TableScanNode scanNode, Captures captures, Context context)
 
         Redirection redirection = getOnlyElement(redirections);
         CatalogSchemaTableName destinationTable = redirection.getDestinationTable();
+
+        QualifiedObjectName destinationObjectName = convertFromSchemaTableName(destinationTable.getCatalogName()).apply(destinationTable.getSchemaTableName());
+        QualifiedObjectName redirectedObjectName = metadata.getRedirectedTableName(context.getSession(), destinationObjectName);
```

Comment:
Redirection in redirection :)
What goes wrong if we allow it or if we ignore it ?

Reply:
I don't think we should do name redirection here.

`applyTableScanRedirect` is an advanced optimization that redirects a scan within a connector, with specific knowledge about the target table, column handles, etc. Why would the connector then choose to redirect

Redirecting to a different catalog would also open up a security hole, since we don't do any security checks at this point -- it would give the source catalog access to something in the target catalog.

Reply:
agreed, we should disallow this

Reply:
current behavior in the PR seems consistent with this, we just do a check here.